libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/MapLayerInteractor.java

```diff
@@ -0,0 +1,109 @@
+package com.mapbox.services.android.navigation.ui.v5.map;
+
+import android.graphics.Bitmap;
+
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.style.layers.Layer;
+import com.mapbox.mapboxsdk.style.layers.LineLayer;
+import com.mapbox.mapboxsdk.style.layers.SymbolLayer;
+import com.mapbox.mapboxsdk.style.sources.Source;
+import com.mapbox.mapboxsdk.style.sources.VectorSource;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static com.mapbox.mapboxsdk.style.layers.Property.NONE;
+import static com.mapbox.mapboxsdk.style.layers.Property.VISIBLE;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.visibility;
+
+class MapLayerInteractor {
+
+  private MapboxMap mapboxMap;
+
+  MapLayerInteractor(MapboxMap mapboxMap) {
+    this.mapboxMap = mapboxMap;
+  }
+
+  void addLayer(Layer layer) {
+    mapboxMap.addLayer(layer);
+  }
+
+  Layer retrieveLayerFromId(String layerId) {
+    return mapboxMap.getLayerAs(layerId);
+  }
+
+  void addLayerImage(String imageName, Bitmap image) {
+    mapboxMap.addImage(imageName, image);
+  }
+
+  void updateLayerVisibility(boolean isVisible, String tileSetIdentifier, String layerIdentifier) {
+    // TODO find sourceIdentifier for Layer https://github.com/mapbox/mapbox-gl-native/issues/12691
+    List<String> sourceIdentifiers = Collections.emptyList();
+    List<Layer> layers = mapboxMap.getLayers();
+    updateLayerWithVisibility(layerIdentifier, sourceIdentifiers, layers, isVisible);
+  }
+
+  boolean isLayerVisible(String tileSetIdentifier, String layerIdentifier) {
+    // TODO find sourceIdentifier for Layer https://github.com/mapbox/mapbox-gl-native/issues/12691
+    List<String> sourceIdentifiers = Collections.emptyList();
+    List<Layer> layers = mapboxMap.getLayers();
+    return findLayerVisibility(layerIdentifier, sourceIdentifiers, layers);
+  }
+
+  // TODO find sourceIdentifier for Layer https://github.com/mapbox/mapbox-gl-native/issues/12691
+  private List<String> findSourceIdentifiersFor(String tileSetIdentifier) {
```

Comment:
This method is unused due to mapbox/mapbox-gl-native#12691. Should it be deleted or commented out until a `getSource()` or `getSourceId()` is available?

Reply:
I went ahead and deleted the unused code but added the changes as a new commit so we'll be able to recover the code from https://github.com/mapbox/mapbox-navigation-android/pull/1226/commits/462bb0a5e8216cfdc9175b04ea0e1f0857e509ca when https://github.com/mapbox/mapbox-gl-native/issues/12691 lands.